packages/hardhat/test/Quest.test.js

```diff
@@ -0,0 +1,152 @@
+const { ethers } = require("hardhat");
+const { fail } = require("assert");
+const { use, expect } = require("chai");
+const { solidity } = require("ethereum-waffle");
+const { deployFakeToken, deployQuest } = require("./test-helper");
+
+const FAKE_ADDRESS = "0x0000000000000000000000000000000000000000";
+
+use(solidity);
+
+describe("[Contract] Quest", function () {
+  let ownerAddress;
+  let playerAddress;
+
+  before(async function () {
+    const [owner, player] = await ethers.getSigners();
+    ownerAddress = owner.address;
+    playerAddress = player.address;
+  });
+
+  describe("recoverUnclaimedFunds()", function () {
+    it("should empty the quest funds", async function () {
+      // Arrange
+      const questFund = 1000;
+      const rewardToken = await deployFakeToken(questFund);
+      const quest = await deployQuest(
+        0x0,
+        rewardToken,
+        1, // Set expireTime to Unix Epoch 0
+        FAKE_ADDRESS,
+        ownerAddress
+      );
+
+      // Act
+      await quest.recoverUnclaimedFunds();
+
+      // Assert
+      expect(await rewardToken.balanceOf(quest.address)).to.eq(0);
+      expect(await rewardToken.balanceOf(ownerAddress)).to.eq(questFund);
+    });
+
+    it("should throw if not expire", async function () {
+      // Arrange
+      const questFund = 1000;
+      const rewardToken = await deployFakeToken(questFund);
+      const quest = await deployQuest(
+        0x0,
+        rewardToken,
+        Math.round(new Date().getTime() / 1000) + 1000, // now in unich epoch
+        FAKE_ADDRESS,
+        ownerAddress
+      );
+
+      try {
+        // Act
+        await quest.recoverUnclaimedFunds();
+        // Assert
+        fail("AssertionError : Should throw because quest is not yet expired");
+      } catch (err) {
+        expect(err.message).to.contains("ERROR: Not expired");
+      }
+    });
+  });
+
+  describe("recoverNativeTokens()", function () {
+    it("should set the balance", async function () {
+      // Arrange
+      const questFund = 1000;
+      const rewardToken = await deployFakeToken(questFund);
+      const quest = await deployQuest(
+        0x0,
+        rewardToken,
+        0,
+        ownerAddress,
+        ownerAddress
+      );
+
+      // Act
+      await quest.recoverNativeTokens();
+
+      // Assert
+      expect(await rewardToken.balanceOf(ownerAddress)).to.eq(0);
+    });
+  });
+
+  describe("claim()", function () {
```

Comment:
What do you think of including a few more tests that:
- Confirm the `QuestClaimed` [emit with the right arguments](https://ethereum-waffle.readthedocs.io/en/latest/matchers.html#emitting-events).
- Assert the transaction reverts if the amount is greater than the quest funds.
- The transaction reverts if the evidence is empty.